solc/CommandLineInterface.cpp

```diff
@@ -614,6 +617,38 @@ bool CommandLineInterface::parseLibraryOption(string const& _input)
 	return true;
 }
 
+//read file contents from m_sourceCodes into m_sourceJsons, which must be a JSON-file as produced by
+// a) --combined-json -ast,compact-format
+// b) the standard-json output
+// and replace the internal map of sourceCodes so that every source has its own entry
+map<string, Json::Value> CommandLineInterface::parseAstFromInput()
+{
+	map<string, Json::Value> sourceJsons;
+	map<string, string> tmpSources;
+	for (auto const& srcPair: m_sourceCodes)
+	{
+		Json::Value ast;
+		if (jsonParse(srcPair.second, ast))
+			if (ast.isMember("sources"))
+				for (auto& src: ast["sources"].getMemberNames())
+				{
+					std::string astKey = ast["sources"][src].isMember("ast") ? "ast" : "AST";
+					astAssert( ast["sources"][src].isMember(astKey), "astkey is not member");
+					astAssert( ast["sources"][src][astKey]["nodeType"].asString() == "SourceUnit",  "Top-level node should be a 'SourceUnit'");
+					astAssert(sourceJsons.count(src) == 0, "All sources must have unique names");
+					sourceJsons.emplace(src, move(ast["sources"][src][astKey]));
+					tmpSources[src] = dev::jsonCompactPrint(ast);
+				}
+			else
+				BOOST_THROW_EXCEPTION(InvalidAstError() << errinfo_comment("Invalid Format for import-JSON: Must have 'sources'-object "));
+		else
+			BOOST_THROW_EXCEPTION(InvalidAstError() << errinfo_comment("Input file could not be parsed to JSON"));
```

Comment:
Use `astAssert` here and above. Basically don't use `BOOST_THROW_EXCEPTION` at all with `InvalidAstError`.